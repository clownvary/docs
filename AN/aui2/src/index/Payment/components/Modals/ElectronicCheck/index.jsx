import PropTypes from 'prop-types';
import React from 'react';
import { getPureNumber } from 'shared/utils/formatNumber';
import InputBankCard from 'react-base-ui/lib/components/InputBankCard';
import Modal from 'react-base-ui/lib/components/Modal';
import Checkbox from 'react-base-ui/lib/components/Checkbox';
import Button from 'react-base-ui/lib/components/Button';
import Dropdown from 'react-base-ui/lib/components/Dropdown';
import Alert from 'react-base-ui/lib/components/Alert';
import Popover from 'react-base-ui/lib/components/Popover';
import UIComponent from 'shared/components/UIComponent';
import AMIds from '../../../automationIds';

import './index.less';

const ACCOUNT_NUMBER = 'accountNumber';
const ROUTING_NUMBER = 'routingNumber';

const ACCOUNT_NUMBER_ERROR = 'Invalid Account Number';
const ROUTING_NUMBER_ERROR = 'Invalid Routing Number';
const defaultState = {
  [ACCOUNT_NUMBER]: '',
  [ROUTING_NUMBER]: ''
};

export default class ECheckModal extends UIComponent {

  static propTypes = {
    data: PropTypes.shape({
      showModel: PropTypes.bool.isRequired,
      accountTypeValue: PropTypes.string.isRequired,
      accountTypeList: PropTypes.shape({
        data: PropTypes.array.isRequired
      }).isRequired,
      saveInformation: PropTypes.bool.isRequired,
      newEcheckError: PropTypes.string.isRequired
    }).isRequired,

    title: PropTypes.string.isRequired,
    showPriorEcp: PropTypes.bool.isRequired,

    onCancel: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    onError: PropTypes.func.isRequired,
    changeSaveAccountInformation: PropTypes.func.isRequired,
    changeAccountType: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      routingNumber: '',
      accountNumber: ''

    };
  }

  componentDidUpdate() {
    const { showModel } = this.props.data;

    if (!showModel) {
      this.clearError();
      this.resetData();
    }
  }

  submit = () => {
    const { accountNumber, routingNumber } = this.state;
    this.props.onSubmit(getPureNumber(accountNumber), getPureNumber(routingNumber));
  }

  changeEcheckAccountTypeHandler = (value) => {
    this.clearError();
    this.props.changeAccountType(value);
  }

  changeValue = (value, fieldName) => {
    const stateObject = {};
    const { newEcheckError } = this.props.data;
    stateObject[fieldName] = value;
    this.setState(stateObject);
    /* istanbul ignore if */
    if ((ROUTING_NUMBER === fieldName && newEcheckError === ROUTING_NUMBER_ERROR)
    || (ACCOUNT_NUMBER === fieldName && newEcheckError === ACCOUNT_NUMBER_ERROR)) {
      this.clearError();
    }
  }

  resetData = () => {
    this.setState({ ...defaultState });
  }

  closeNewECheck() {
    this.props.onCancel();
  }

  clearError = () => {
    if (this.props.data.newEcheckError) {
      this.props.onError('');
    }
  }

  render() {
    const {
      showModel,
      accountTypeValue,
      accountTypeList,
      saveInformation,
      newEcheckError
    } = this.props.data;
    const { accountNumber, routingNumber } = this.state;

    return (
      <div>
        <Modal
          title={this.props.title}
          shown={showModel}
          className="new-eCheck-modal"
          onClose={() => this.closeNewECheck()}
        >
          <div className="modal-body">
            {/* Error messages */}
            {
              /* eslint-disable react/jsx-boolean-value */
              newEcheckError ?
                <Alert type="danger" noClose>
                  <ul className="only-one-error">
                    <li>{newEcheckError}</li>
                  </ul>
                </Alert> : ''
              /* eslint-enable react/jsx-boolean-value */
            }
            <div className="form-group">

              {/* accountNumber row */}
              <div className={`form-row aaui-flexbox ${newEcheckError === ACCOUNT_NUMBER_ERROR ? 'form-error' : ''}`}>
                <label className="form-label-col" htmlFor={ACCOUNT_NUMBER}>
                  Account Number
                </label>
                <div className="form-entry-col">
                  {
                    showModel &&
                    <InputBankCard
                      id={ACCOUNT_NUMBER}
                      value={accountNumber}
                      maxLength={50}
                      data-qa-id={AMIds.modals.accountNumber}
                      onInput={(e, value) => { this.changeValue(value, ACCOUNT_NUMBER); }}
                    />
                  }
                </div>
                <div className="form-tooltip-col">
                  <span className="icon icon-info-circle popover-base" data-popover-trigger>
                    <Popover className="tooltip">
                      <div className="new_echeck_tip" />
                    </Popover>
                  </span>
                </div>
              </div>

              {/* routingNumber row */}
              <div className={`form-row  aaui-flexbox ${newEcheckError === ROUTING_NUMBER_ERROR ? 'form-error' : ''}`}>
                <label className="form-label-col" htmlFor={ROUTING_NUMBER}>
                  Routing Number
                </label>
                <div className="form-entry-col">
                  {showModel &&
                    <InputBankCard
                      id={ROUTING_NUMBER}
                      maxLength={10}
                      value={routingNumber}
                      data-qa-id={AMIds.modals.routingNumber}
                      onInput={(e, value) => { this.changeValue(value, ROUTING_NUMBER); }}
                    />
                  }
                </div>
                <div className="form-tooltip-col">
                  <span className="icon icon-info-circle popover-base" data-popover-trigger>
                    <Popover className="tooltip">
                      <div className="new_echeck_tip" />
                    </Popover>
                  </span>
                </div>
              </div>

              {/* echeck Type row */}
              {
                <div className="form-row aaui-flexbox">
                  <label className="form-label-col" htmlFor="accountType">
                    Account Type
                  </label>
                  <div className="form-entry-col">
                    <Dropdown
                      id="accountType"
                      value={accountTypeValue}
                      data-qa-id={AMIds.modals.accountType}
                      data={accountTypeList.data}
                      placeholder="Select Account type"
                      onChange={({ value }) => this.changeEcheckAccountTypeHandler(value)}
                      className="echeck-type-dropdown"
                    />
                  </div>
                </div>
              }

              {/* save echeck row */}
              <div className="form-row aaui-flexbox">
                <label className="form-label-col" htmlFor="newECheckCheckBox" />
                <div className="form-entry-col">
                  {
                    this.props.showPriorEcp ?
                      <Checkbox
                        id="newECheckCheckBox"
                        data-qa-id={AMIds.modals.saveInformation}
                        size="m"
                        checked={saveInformation}
                        onChange={e =>
                          this.props.changeSaveAccountInformation(e.target.checked)
                        }
                      >
                        Save account information
                    </Checkbox> : ''
                  }
                </div>
              </div>
            </div>
          </div>
          <div className="modal-footer">
            <Button onClick={() => this.closeNewECheck()}>Cancel</Button>
            <Button
              type="strong"
              onClick={this.submit}
              disabled={!accountNumber || !routingNumber || newEcheckError}
            >
              OK
            </Button>
          </div>
        </Modal>
      </div>
    );
  }
}
