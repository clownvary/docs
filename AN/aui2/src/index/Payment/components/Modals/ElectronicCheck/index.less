@import '~shared/style/variables/index.less';
/* stylelint-disable */

@image-url: '../../../../../shared/images';

.new-eCheck-modal {
  .echeck-type-dropdown {
    width: 180px;
  }

  .alert {
    margin-bottom: 30px;

    .only-one-error {
      display: inline-block;
      width: 520px;
      padding-left: 10px;
      margin: 0;
      line-height: 1.4;
      vertical-align: top;
      list-style: none;
    }
  }

  .form-group {
    .form-row {
      align-items: flex-start;
      min-height: 34px;
      margin: 10px 0 0;

      &:frist-child {
        margin: 0;
      }
    }

    .form-error {
      .checkbox__text:before,
      .input-group > input,
      .aaui-dropdown > button,
      .aaui-dropdown .aaui-dropdown__button-text {
        color: @state-danger-text;
        background-color: @state-danger-bg;
        border-color: @state-danger-border;
      }

      .checkbox {
        color: @state-danger-text;
      }

      .form-label-col {
        color: @state-danger-text;
      }
    }

    .form-label-col {
      flex: 0 0 141px;
      padding-right: 16px;
      margin-top: 2px;
      font-weight: @semibold;
      text-align: right;
    }

    .form-entry-col {
      flex: 0 0 320px;
      box-sizing: border-box;
    }

    .form-tooltip-col {
      flex: 0 0 36px;
      text-align: center;
    }

    .input-group {
      display: inline;
      width: 320px;
    }

    p {
      margin: 0;
      font-size: 14px;
      color: #db3811;
    }
  }

  .popover-base {
    margin-top: 6px;
    color: @color-blue;

    .popover--e {
      transform: translate(0, -20px);

      &:before,
      &:after {
        top: 20px;
      }
    }
  }

  .new_echeck_tip {
    width: 237px;
    height: 122px;
    background: url('@{image-url}/img_echeck_tips.png') top center no-repeat;
    background-size: 237px 122px;
  }
}
