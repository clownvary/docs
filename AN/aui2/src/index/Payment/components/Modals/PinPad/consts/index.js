import * as titles from './titles';
import * as actionTypes from './actionTypes';

export default {
  titles,
  actionTypes
};
