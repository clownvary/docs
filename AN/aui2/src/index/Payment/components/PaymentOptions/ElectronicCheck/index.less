.payment-electronic-check {
  .electronic-check-dropdown {
    margin: 0 10px;

    .aaui-dropdown__menu {
      ul {
        max-height: none;
      }

      li {
        padding: 0 1em;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }
  }

  .aaui-flexbox {
    justify-content: space-between;
  }
}

.permit-btn-group {
  display: flex;
  justify-content: space-between;
  padding-top: 60px;
}
