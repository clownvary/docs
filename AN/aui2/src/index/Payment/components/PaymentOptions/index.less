@import '~shared/style/variables/index.less';

@option-split-border: solid 2px #d6d7da;
@option-split-interval: 18px;
@input-width: 112px;
@font-bold: 600;
/* stylelint-disable */

.payment {
  &-options {
    h2 {
      padding-bottom: 10px;
      margin-top: 0;
      font-size: 20px;
      line-height: 1.5;
      text-transform: capitalize;
      color: #313741;
    }

    .payment {
      &-action {
        justify-content: space-between;
        padding-bottom: 8px;
        font-size: 14px;
      }

      &-distribution {
        .checkbox__text {
          font-size: 14px;
        }
      }

      &-title {
        padding-left: 11px;
        text-transform: uppercase;
        color: #748288;
      }

      &-splition {
        padding-right: 4px;
        cursor: pointer;
        color: @color-blue;

        .icon {
          padding-right: 5px;
        }

        &-disabled {
          cursor: default;
          color: #9b9b9b;

          .icon {
            padding-right: 5px;
            cursor: default;
          }
        }
      }
    }

    .input-group {
      box-sizing: border-box;
      width: @input-width;
    }

    .payment-paynow {
      margin-bottom: 15px;

      .paynow-label {
        display: inline-block;
        width: 114px;
        padding-right: 20px;
        font-weight: @font-bold;
        text-align: right;
      }

      .input-group {
        display: inline-table;
        width: 185px;
        margin-right: 25px;
        vertical-align: middle;
      }

      .paymentplan-label {
        padding-right: 20px;
        font-weight: @font-bold;
      }
    }

    .paynow-alert-error,
    .paymentplan__alert--error {
      .modal-close {
        display: none;
      }
    }
  }

  &-list {
    border-top: @option-split-border;

    & > div:last-child .payment-option {
      margin-bottom: 0;
    }
  }

  &-list-new {
    display: flex;
    flex-direction: column-reverse;
    justify-content: flex-start;
    border-top: @option-split-border;

    &__payment-plan-item {
      flex-grow: 1;
      flex-shrink: 1;
      order: -1;
    }

    &__item {
      flex-grow: 1;
      flex-shrink: 1;
    }
  }

  &-option {
    border-bottom: @option-split-border;

    &-list {
      padding: @option-split-interval 10px 14px 10px;

      &--error {
        z-index: 1;
        background: @state-danger-bg;
        border: 1px solid @state-danger-border;
      }

      &.aaui-flexbox {
        align-items: flex-start;
      }

      .aaui-dropdown {
        width: 170px;
      }
    }

    .icon-trash {
      padding-left: 10px;
      text-align: center;
      cursor: pointer;
      color: @color-blue;

      &-disabled {
        cursor: default;
        color: #9b9b9b;
      }

      &-paymentplan {
        visibility: hidden;
      }
    }
  }

  &-amount-exception-alert {
    .modal-close {
      display: none;
    }
  }

  &-new-cart-msg {
    padding-left: 10px;
    font-size: 12px;
    color: #748288;
  }

  .aaui-flexbox {
    display: flex;
  }
}
