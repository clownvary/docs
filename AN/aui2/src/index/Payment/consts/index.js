import creditCardTypes from './ccTypes';
import * as frequencyTypes from './frequencyTypes';
import * as payerConfig from './payerConfig';
import * as paymentPlanPaymentTypes from './paymentPlanPaymentTypes';
import * as paymentTypes from './paymentTypes';
import * as paymentTypeNames from './paymentTypeNames';
import * as newOptions from './newOptions';
import * as actionTypes from './actionTypes';
import * as fieldProps from './fieldProps';
import * as CCPaymentMethods from './CCPaymentMethods';
import * as resetTypes from './resetTypes';
import cardTypesEnum from './cardTypes';

export default {
  creditCardTypes,
  frequencyTypes,
  payerConfig,
  paymentPlanPaymentTypes,
  paymentTypes,
  paymentTypeNames,
  newOptions,
  actionTypes,
  fieldProps,
  cardTypesEnum,
  CCPaymentMethods,
  resetTypes
};
