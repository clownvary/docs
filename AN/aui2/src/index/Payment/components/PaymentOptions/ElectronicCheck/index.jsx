import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import InputNumeric from 'react-base-ui/lib/components/InputNumeric';

import Dropdown from 'react-base-ui/lib/components/Dropdown';
import { addError } from 'shared/actions/Error';
import UIComponent from 'shared/components/UIComponent';

import { changeRemaining } from '../../../actions';
import { getDefaultAmount } from '../utils/payment';
import checkError from '../utils/checkError';

import {
  changeECheckAmount,
  changeElectronicCheckOpiton,
  fetchECheckConfig
} from '../../../actions/paymentOptions/electronicCheck';
import { newOptions, fieldProps } from '../../../consts';
import AMIds from '../../../automationIds';

import './index.less';

export class ElectronicCheck extends UIComponent {
  static propTypes = {
    item: PropTypes.shape({
      amount: PropTypes.string.isRequired
    }),

    /* eslint-disable react/forbid-prop-types */
    index: PropTypes.number.isRequired,
    data: PropTypes.object.isRequired,
    value: PropTypes.object.isRequired,
    children: PropTypes.object.isRequired,
    isRefund: PropTypes.bool.isRequired,
    newECheck: PropTypes.object.isRequired,
    optionLen: PropTypes.number.isRequired,
    /* eslint-enable react/forbid-prop-types */

    changeECheckAmount: PropTypes.func.isRequired,
    changeRemaining: PropTypes.func.isRequired,
    fetchECheckConfig: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    changeElectronicCheckOpiton: PropTypes.func.isRequired,
    addError: PropTypes.func.isRequired
  }


  constructor(props) {
    super(props);
    this.state = {
      amount: props.item.amount,
      overrideMsg: ''
    };
  }

  componentWillMount() {
    const eCheckList = this.props.data.eCheckListDropDown.data;

    // thie data have a defalut use-new-electronic-card option
    if (!eCheckList.length ||
      (eCheckList.length === 1 && eCheckList[0].value === newOptions.NEW_OPTION_VALUE)) {
      this.props.fetchECheckConfig();
    }
  }

  changeAmountAndRemaining(hasError) {
    const { index, isRefund } = this.props;
    let amount = parseFloat(this.state.amount || 0);
    const availableAmount = +getDefaultAmount(index);

    hasError && this.props.clearOptionAndPaymentErrs(index);

    if (availableAmount < amount) {
      amount = availableAmount;
      const message = isRefund ?
        fieldProps.MAX_AMOUNT_MESSAGE_ON_REFUND :
        fieldProps.MAX_AMOUNT_MESSAGE_ON_NOT_REFUND;
      this.props.addError({
        payload: {
          code: -1,
          message
        }
      });
    }
    this.saveECheckCardInfo({ amount, key: index });
  }

  saveECheckCardInfo({ amount, key }) {
    const defaultValue = +getDefaultAmount(key);
    const remaining = Math.max(defaultValue - amount);
    /* istanbul ignore else */
    if (remaining >= 0) {
      const formatECheckAmount = amount.toFixed(2);

      this.setState({ amount: formatECheckAmount });
      this.props.changeECheckAmount({ key, amount: formatECheckAmount, formatECheckAmount });
      this.props.changeRemaining({ remaining: remaining.toFixed(2) });
    }
  }

  render() {
    const { item, value, index, children,
      optionLen, data, hasManuallySplitOption } = this.props;
    const {
      eCheckLabel,
      eCheckListDropDown
    } = data;
    const { eCheckListDropDownValue } = item;
    const hasError = checkError(item);
    const hasCardListError = checkError(item, 'echeck');

    return (
      <div className="payment-option payment-electronic-check">
        <div
          className={classNames(
            'payment-option-list',
            'aaui-flexbox',
            { 'payment-option-list--error': hasError }
          )}
        >
          <div className="aaui-flexbox aaui-flexbox-c1">
            <Dropdown
              data={item.list}
              value={value}
              ref={(input) => { this._searchInput = input; }}
              data-qa-id={AMIds.paymentOption.optionList}
              onChange={({ value: val }) => { this.props.onChange(val); }}
              disabled={index !== optionLen - 1 && hasManuallySplitOption}
            />

            <Dropdown
              placeholder={`Choose ${eCheckLabel}`}
              errored={hasCardListError}
              value={eCheckListDropDownValue}
              data={eCheckListDropDown.data}
              /* eslint-disable react/jsx-boolean-value */
              showTextTip={true}
              /* eslint-enable react/jsx-boolean-value */
              className="electronic-check-dropdown"
              data-qa-id={AMIds.paymentOption.echeckList}
              onChange={({ value: val }) =>
                this.onChangeECP(index, val, hasError || hasCardListError)}
            />

          </div>

          <div className="aaui-flexbox">
            <label className="payment-symbol" htmlFor="paymentAmount">$</label>
            <InputNumeric
              id="paymentAmount"
              value={this.state.amount}
              data-qa-id={AMIds.paymentOption.optionAmount}
              min={0}
              onValueChange={e => this.setState({
                amount: e.value
              })}
              onBlur={() => this.changeAmountAndRemaining(hasError)}
            />
            {children}
          </div>
        </div>

      </div>
    );
  }

  onChangeECP(index, val, clearErr) {
    if (clearErr) {
      this.props.clearOptionAndPaymentErrs(index);
    }
    this.props.changeElectronicCheckOpiton(index, val);
  }
}

export default connect(
  null,
  {
    changeECheckAmount,
    changeElectronicCheckOpiton,
    changeRemaining,
    fetchECheckConfig,
    addError
  }
)(ElectronicCheck);
