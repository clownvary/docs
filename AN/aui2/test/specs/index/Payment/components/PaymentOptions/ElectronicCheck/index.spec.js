import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { ElectronicCheck } from 'index/Payment/components/PaymentOptions/ElectronicCheck';
import InputNumeric from 'react-base-ui/lib/components/InputNumeric';
import Dropdown from 'react-base-ui/lib/components/Dropdown';
import { newOptions } from 'index/Payment/consts';
import resetActions from 'utils/resetActions';

jest.mock('index/Payment/components/PaymentOptions/utils/payment', () => ({
  getDefaultAmount: jest.fn().mockReturnValue(300)
}));

describe('index/Payment/components/PaymentOptions/ElectronicCheck', () => {
  const actions = {
    changeRemaining: jest.fn(),
    changeECheckAmount: jest.fn(),
    changeElectronicCheckOpiton: jest.fn(),
    fetchECheckConfig: jest.fn(),
    addError: jest.fn(),
    onChange: jest.fn(),
    clearOptionAndPaymentErrs: jest.fn()
  };

  afterEach(() => {
    resetActions(actions);
  });

  const setup = (props) => mount(<ElectronicCheck {...props} />);

  it('ElectronicCheck component should render without error', () => {
    const props = {
      item: {
        list: [
          { value: '173', name: 'ElectronicCheck' },
          { value: '172', name: 'Credit' }
        ],
        eCheckListDropDownValue: '445',
        disabled: false,
        amount: '500'
      },
      value: '173',
      index: 0,
      optionLen: 3,
      children: (<span>ElectronicCheck info</span>),
      data: {
        eCheckLabel: 'Electronic Check',
        eCheckListDropDown: {
          data: [
            { value: '443', name: 'electronic #2', text: 'electronic #2' },
            { value: '444', name: 'electronic #3', text: 'electronic #3' },
            { value: '445', name: 'electronic #5', text: 'electronic #5' }
          ]
        }
      },
      isRefund: false,
      ...actions
    };

    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('ElectronicCheck component change payment option should work fine', () => {
    const props = {
      item: {
        list: [
          { value: '173', name: 'ElectronicCheck' },
          { value: '172', name: 'Credit' }
        ],
        eCheckListDropDownValue: '',
        disabled: false,
        amount: '500'
      },
      value: '173',
      index: 0,
      optionLen: 3,
      children: (<span>ElectronicCheck info</span>),
      data: {
        eCheckLabel: 'Electronic Check',
        eCheckListDropDown: {
          data: [
            { value: newOptions.NEW_OPTION_VALUE, name: 'select electronic check', text: 'select electronic check' }
          ]
        }
      },
      isRefund: false,
      ...actions
    };

    const component = setup(props);
    const instance = component.instance();

    expect(instance.state.amount).toEqual('500');
    expect(instance.state.overrideMsg).toEqual('');
    expect(component.find('.payment-electronic-check')).toHaveLength(1);
    expect(component.find('#paymentAmount')).toHaveLength(1);
    expect(actions.fetchECheckConfig).toHaveBeenCalled();

    component.find(Dropdown).at(0).node.props.onChange({});
    expect(actions.onChange).toBeCalled();

    component.find(Dropdown).at(1).node.props.onChange({});
    expect(actions.changeElectronicCheckOpiton).toBeCalled();
  });

  it('ElectronicCheck component change credit card option and amount should work fine', () => {
    const props = {
      item: {
        list: [
          { value: '173', name: 'ElectronicCheck' },
          { value: '172', name: 'Credit' }
        ],
        eCheckListDropDownValue: '445',
        disabled: false,
        errors: [{ key: 0, name: '' }]
      },
      value: '173',
      index: 0,
      optionLen: 3,
      children: (<span>ElectronicCheck info</span>),
      data: {
        eCheckLabel: 'Electronic Check',
        eCheckListDropDown: {
          data: [
            { value: '443', name: 'electronic #2', text: 'electronic #2' },
            { value: '444', name: 'electronic #3', text: 'electronic #3' },
            { value: '445', name: 'electronic #5', text: 'electronic #5' }
          ]
        }
      },
      ...actions
    };

    const component = setup(props);
    const instance = component.instance();

    const amountInput = component.find(InputNumeric);
    const amountInputHtmlElement = component.find('#paymentAmount');
    amountInput.node.input.value = 0;
    amountInput.node.props.onValueChange({ value: 0 });
    amountInputHtmlElement.simulate('blur');
    expect(actions.addError).not.toHaveBeenCalled();
    expect(actions.changeECheckAmount).toHaveBeenCalledTimes(1);
    expect(actions.changeRemaining).toHaveBeenCalledTimes(1);
    expect(instance.state.amount).toEqual('0.00');

    amountInput.node.input.value = 400;
    amountInput.node.props.onValueChange({ value: 400 });
    amountInputHtmlElement.simulate('blur');
    expect(actions.addError).toHaveBeenCalledTimes(1);
    expect(actions.changeECheckAmount).toHaveBeenCalledTimes(2);
    expect(actions.changeRemaining).toHaveBeenCalledTimes(2);
    expect(instance.state.amount).toEqual(300);

    component.setProps({ isRefund: true });
    amountInput.node.props.onValueChange({ value: 400 });
    amountInputHtmlElement.simulate('blur');
    expect(actions.addError).toHaveBeenCalledTimes(2);
    expect(actions.changeECheckAmount).toHaveBeenCalledTimes(3);
    expect(actions.changeRemaining).toHaveBeenCalledTimes(3);
    expect(instance.state.amount).toEqual(300);
    component.find(Dropdown).at(1).node.props.onChange({});
    expect(actions.clearOptionAndPaymentErrs).toBeCalled();
  });
});
