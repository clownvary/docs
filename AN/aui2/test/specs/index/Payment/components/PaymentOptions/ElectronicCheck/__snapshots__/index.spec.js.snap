// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/Payment/components/PaymentOptions/ElectronicCheck ElectronicCheck component should render without error 1`] = `
<ElectronicCheck
  addError={[Function]}
  changeECheckAmount={[Function]}
  changeElectronicCheckOpiton={[Function]}
  changeRemaining={[Function]}
  clearOptionAndPaymentErrs={[Function]}
  data={
    Object {
      "eCheckLabel": "Electronic Check",
      "eCheckListDropDown": Object {
        "data": Array [
          Object {
            "name": "electronic #2",
            "text": "electronic #2",
            "value": "443",
          },
          Object {
            "name": "electronic #3",
            "text": "electronic #3",
            "value": "444",
          },
          Object {
            "name": "electronic #5",
            "text": "electronic #5",
            "value": "445",
          },
        ],
      },
    }
  }
  fetchECheckConfig={[Function]}
  index={0}
  isRefund={false}
  item={
    Object {
      "amount": "500",
      "disabled": false,
      "eCheckListDropDownValue": "445",
      "list": Array [
        Object {
          "name": "ElectronicCheck",
          "value": "173",
        },
        Object {
          "name": "Credit",
          "value": "172",
        },
      ],
    }
  }
  onChange={[Function]}
  optionLen={3}
  value="173"
>
  <div
    className="payment-option payment-electronic-check"
  >
    <div
      className="payment-option-list aaui-flexbox"
    >
      <div
        className="aaui-flexbox aaui-flexbox-c1"
      >
        <Dropdown
          autoOpen={false}
          data={
            Array [
              Object {
                "name": "ElectronicCheck",
                "value": "173",
              },
              Object {
                "name": "Credit",
                "value": "172",
              },
            ]
          }
          data-qa-id="payment-paymentOption-optionList"
          filterPlaceholder="Filter..."
          fuzzyQuery={false}
          noResult="No results found."
          onChange={[Function]}
          placeholder="Select one..."
          showAll={true}
          showDeselectall={false}
          showTxtOnlyCheckedOneItem={true}
          theme="flat"
          txtSuffix="selected"
          value="173"
          visible={true}
        >
          <div
            className="aaui-dropdown"
            data-qa-id="payment-paymentOption-optionList"
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
          >
            <button
              className="aaui-dropdown__button aaui-dropdown-btn-flat"
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="button"
              type="button"
            >
              <div
                className="aaui-dropdown__button-text"
                title=""
              >
                 Select one...
              </div>
              <i
                className="icon aaui-dropdown__button-icon icon-chevron-down"
              />
            </button>
            <div
              aria-expanded={false}
              className="aaui-dropdown__menu-wrapper  u-hidden"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={0}
            >
              <div
                className="aaui-dropdown__menu"
              >
                <ul
                  aria-expanded={false}
                  className=""
                  role="listbox"
                >
                  <li
                    className=" is-selected"
                    key="173"
                    onClick={[Function]}
                    role="option"
                  >
                    <span />
                  </li>
                  <li
                    className=""
                    key="172"
                    onClick={[Function]}
                    role="option"
                  >
                    <span />
                  </li>
                </ul>
                <div>
                  <div
                    className="aaui-dropdown__menu-footer"
                  >
                    <div
                      className="u-hidden"
                    >
                      <i
                        className="icon icon-spinner"
                      />
                       Loading...
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Dropdown>
        <Dropdown
          autoOpen={false}
          className="electronic-check-dropdown"
          data={
            Array [
              Object {
                "name": "electronic #2",
                "text": "electronic #2",
                "value": "443",
              },
              Object {
                "name": "electronic #3",
                "text": "electronic #3",
                "value": "444",
              },
              Object {
                "name": "electronic #5",
                "text": "electronic #5",
                "value": "445",
              },
            ]
          }
          data-qa-id="payment-paymentOption-echeckList"
          errored={false}
          filterPlaceholder="Filter..."
          fuzzyQuery={false}
          noResult="No results found."
          onChange={[Function]}
          placeholder="Choose Electronic Check"
          showAll={true}
          showDeselectall={false}
          showTextTip={true}
          showTxtOnlyCheckedOneItem={true}
          theme="flat"
          txtSuffix="selected"
          value="445"
          visible={true}
        >
          <div
            className="aaui-dropdown electronic-check-dropdown"
            data-qa-id="payment-paymentOption-echeckList"
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
          >
            <button
              aria-label="electronic #5"
              className="aaui-dropdown__button aaui-dropdown-btn-flat"
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="button"
              type="button"
            >
              <div
                className="aaui-dropdown__button-text"
                title=" electronic #5"
              >
                 electronic #5
              </div>
              <i
                className="icon aaui-dropdown__button-icon icon-chevron-down"
              />
            </button>
            <div
              aria-expanded={false}
              className="aaui-dropdown__menu-wrapper  u-hidden"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={0}
            >
              <div
                className="aaui-dropdown__menu"
              >
                <ul
                  aria-expanded={false}
                  className=""
                  role="listbox"
                >
                  <li
                    className=""
                    key="443"
                    onClick={[Function]}
                    role="option"
                    title="electronic #2"
                  >
                    <span>
                      electronic #2
                    </span>
                  </li>
                  <li
                    className=""
                    key="444"
                    onClick={[Function]}
                    role="option"
                    title="electronic #3"
                  >
                    <span>
                      electronic #3
                    </span>
                  </li>
                  <li
                    className=" is-selected"
                    key="445"
                    onClick={[Function]}
                    role="option"
                    title="electronic #5"
                  >
                    <span>
                      electronic #5
                    </span>
                  </li>
                </ul>
                <div>
                  <div
                    className="aaui-dropdown__menu-footer"
                  >
                    <div
                      className="u-hidden"
                    >
                      <i
                        className="icon icon-spinner"
                      />
                       Loading...
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Dropdown>
      </div>
      <div
        className="aaui-flexbox"
      >
        <label
          className="payment-symbol"
          htmlFor="paymentAmount"
        >
          $
        </label>
        <InputNumeric
          allowBlank={true}
          allowKeySpin={true}
          allowMouseSpin={false}
          className=""
          currency="USD"
          data-qa-id="payment-paymentOption-optionAmount"
          decimals={2}
          disabled={false}
          icon=""
          iconHint=""
          id="paymentAmount"
          increment={1}
          items={Array []}
          listClass="input-numeric-list"
          listClassName=""
          listConfig={Object {}}
          listMaxHeight="300px"
          listPopupOptions={Object {}}
          listWidth="auto"
          max={9999999999.99}
          min={0}
          name=""
          onBlur={[Function]}
          onFocus={[Function]}
          onListClose={[Function]}
          onListOpen={[Function]}
          onTextChange={[Function]}
          onValueChange={[Function]}
          placeHolder=""
          prefix="an"
          showClear={false}
          showGroup={false}
          showSpinner={false}
          showTrigger={false}
          showTrigger2={false}
          size="md"
          textAlign="right"
          triggerIcon="icon-chevron-down"
          triggerIcon2="icon-chevron-down"
          triggerIconHint=""
          triggerIconHint2=""
          triggerIconToggle="icon-chevron-up"
          triggerIconToggle2="icon-chevron-up"
          value="500"
        >
          <div
            className="an-input-numeric input-group buttons-0 input-group--md"
          >
            <input
              className="input input-group__field textalign-right input__field"
              data-qa-id="payment-paymentOption-optionAmount"
              disabled={false}
              id="paymentAmount"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onDragEnter={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              onPaste={[Function]}
              placeholder=""
              type="text"
            />
          </div>
        </InputNumeric>
        <span>
          ElectronicCheck info
        </span>
      </div>
    </div>
  </div>
</ElectronicCheck>
`;
